### PR TITLE
Feature/wallet token freeze unfreeze

### PR DESCRIPTION
**Pull request by OBorce**
- add new commands for freeze and unfreeze tokens
- check send/mint/unmint of tokens if they are frozen before creating the transaction
- add tests

### CODE REVIEW

**Review by azarovh — APPROVED** [review]

File: wallet/src/account/output_cache/mod.rs
Comment (TheQuantumPhysicist) [inline wallet/src/account/output_cache/mod.rs:225]: If possible, please always use False=0, and True=1. In many instances I see the opposite, which is confusing for no good reason. For example the IsTokenUnfreezable enum.

File: wallet/src/wallet/mod.rs
Comment (TheQuantumPhysicist) [inline wallet/src/wallet/mod.rs:157]: I think the word Unfreezable is ambiguous. It can mean two things:

1. A token that on issuance, was set to not be freezable (I would call that NonFreezable)
2. A token that was frozen, but can be unfrozen.

I would recommend disambiguating those cases by using the word NonFreezable where appropriate.

File: wallet/wallet-controller/src/synced_controller.rs
Comment (TheQuantumPhysicist) [inline wallet/wallet-controller/src/synced_controller.rs:532]: frozen

File: wallet/src/wallet/tests.rs
Comment (TheQuantumPhysicist) [inline wallet/src/wallet/tests.rs:2121]: unfreeze

**Review by TheQuantumPhysicist — APPROVED** [review]
Besides the minor comments, looking good.

Please merge this only after all the consensus stuff is merged, followed by a rebase.

File: wallet/src/account/output_cache/mod.rs
Comment (OBorce) [reply wallet/src/account/output_cache/mod.rs:225]: changed the order

**Pull request merged**
